core/field_note.js

```diff
@@ -42,809 +35,797 @@ goog.require('goog.userAgent');
-  if (absDiff < 1) {
-    this.pianoSVG_.setAttribute('transform', 'translate(0, 0)');
-    this.setCKeyLabelsVisible_(true);
-    this.playNoteInternal_();
-    return;
+    this.animationTarget_ = this.fieldEditorWidth_ * octaves * -1;
+    this.animationPos_ = 0;
+    this.stepOctaveAnimation_();
+    this.setCKeyLabelsVisible_(false);
   }
-  this.animationPos_ += (this.animationTarget_ - this.animationPos_) *
-    Blockly.FieldNote.ANIMATION_FRACTION;
-  this.pianoSVG_.setAttribute('transform', 'translate(' + this.animationPos_ + ',0)');
-  requestAnimationFrame(this.stepOctaveAnimation_.bind(this));
-};
 
-/**
- * Set the selected note number, and update the piano display and the input field.
- * @param {number} noteNum The MIDI note number to select.
- * @private
- */
-Blockly.FieldNote.prototype.setNoteNum_ = function(noteNum) {
-  noteNum = this.callValidator(noteNum);
-  this.setValue(noteNum);
-  Blockly.FieldTextInput.htmlInput_.value = noteNum;
-};
-
-/**
- * Sets the text in this field.  Triggers a rerender of the source block, and
- * updates the selection on the field.
- * @param {?string} text New text.
- */
-Blockly.FieldNote.prototype.setText = function(text) {
-  Blockly.FieldNote.superClass_.setText.call(this, text);
-  if (!this.textElement_) {
-    // Not rendered yet.
-    return;
+  /**
+   * Animate the piano up or down an octave by sliding it to the left or right.
+   * @private
+   */
+  stepOctaveAnimation_() {
+    var absDiff = Math.abs(this.animationPos_ - this.animationTarget_);
+    if (absDiff < 1) {
+      this.pianoSVG_.setAttribute('transform', 'translate(0, 0)');
+      this.setCKeyLabelsVisible_(true);
+      this.playNoteInternal_();
+      return;
+    }
+    this.animationPos_ += (this.animationTarget_ - this.animationPos_) *
+      FieldNote.ANIMATION_FRACTION;
+    this.pianoSVG_.setAttribute('transform', 'translate(' + this.animationPos_ + ',0)');
+    requestAnimationFrame(this.stepOctaveAnimation_.bind(this));
   }
-  this.updateSelection_();
-  // Cached width is obsolete.  Clear it.
-  this.size_.width = 0;
-};
 
-/**
- * For a MIDI note number, find the index of the corresponding piano key.
- * @param {number} noteNum The note number.
- * @returns {number} The index of the piano key.
- * @private
- */
-Blockly.FieldNote.prototype.noteNumToKeyIndex_ = function(noteNum) {
-  return Math.floor(noteNum) - (this.displayedOctave_ * 12);
-};
+  /**
+   * Set the selected note number, and update the piano display and the input field.
+   * @param {number} noteNum The MIDI note number to select.
+   * @private
+   */
+  setNoteNum_(noteNum) {
+    noteNum = this.doClassValidation_(noteNum);
+    this.setValue(noteNum);
+    this.htmlInput_.value = noteNum;
+
+    if (!this.textElement_) {
+      // Not rendered yet.
+      return;
+    }
+    this.updateSelection_();
+    // Cached width is obsolete.  Clear it.
+    this.size_.width = 0;
+  }
```

Comment:
Can you remove this and just have it use `setValue` + override the other value methods? I think that should be workable.

Reply:
Done, had to use `setEditorValue_` to have the change reflected in the block though.

Reply:
Yeah that should be correct! I disagreed with that design decision, but it is how it works now :P lmk if it needs doc updates.